docs/en/join-testnet.md

```diff
@@ -4,58 +4,168 @@ title: Join Testnets
 
-**Prerequisite**
+## Overview
 
- - Loom installed. See [instructions](basic-install-all.html).
+Deploying to Loom PlasmaChain is no different from deploying to the Ethereum main net, or to Rinkeby. If you know how to do one, you also know how to do the other.
 
-### Generate a private key
+We at Loom are using Truffle to build, test, and deploy our smart contracts. To make our life easier, we developed something called a provider that lets Truffle deploy to Loom PlasmaChain just like it deploys to Ethereum.
 
-Once Loom is installed, you need to **generate a private key with Loom and get your public address**:
+Without delving too much into details, the provider works like a bridge that makes Web3.js calls compatible with Loom PlasmaChain.
+
+This document shows how you can use `Truffle` and the [Loom Truffle Provider](https://github.com/loomnetwork/loom-truffle-provider) to build a simple Web UI that interacts with Loom PlasmaChain.
+
+## Requirements
+
+Make sure the following requirements are met and then follow the steps in the next section:
+
+```text
+Node >= 8
+yarn or npm
+```
+
+## Setting Things Up
+
+First, let's clone the [Truffle DAppChainExample](https://github.com/loomnetwork/truffle-dappchain-example/) repository. Open a terminal, `cd` into the directory where you store your projects, and run the following command:
+
+```bash
+git clone https://github.com/loomnetwork/truffle-dappchain-example
+```
+
+Next, `cd` into `truffle-dappchain-example`:
+
+```bash
+cd truffle-dappchain-example
+```
+
+and run:
+
+```bash
+yarn install
+```
+
+## Run against Loom Testnet
+
+### Generate the Loom private key
+
+The following command will download the `loom` binary and generate the private key for you:
+
+```bash
+yarn gen:extdev-key
+```
+
+> The Loom private key will be saved into a file called `extdev_private_key`.
+
+
+### Deploy to extdev_plasma_us1
+
+As an example, we will deploy and run our application against `extdev_plasma_us1`.
+
+Run the following command:
+
+```bash
+yarn deploy:extdev
+```
+
+>The above command will **compile and then deploy** our smart contract to `extdev_plasma_us1`
+
+
+## Testing
+
+Run this command below to send transactions to the smart contract deployed to `extdev_plasma_us1`:
+
+
+```test
+yarn test:extdev
+```
+
+If everything looks good, let's spin up a web server and interact with our smart contract.
+
+## Start the Web Interface
+
+The web interface is built with React and Webpack. Open a new terminal and run the following command:
+
+```bash
+yarn serve:extdev
+```
+
+> The web interface is available on http://localhost:8080.
+
+
+## Run Against Local Loom Chain
+
+First, you have to generate a private key using:
+
+```bash
+yarn gen:loom-key
+```
+
+### Spin up Loom Chain
+
+In a new terminal, run:
 
 ```bash
-./loom genkey -k priv_key -a pub_key
+yarn loom-init
```

Comment:
Does this new command require loom?

Reply:
> Does this new command require loom?

Yes, it initializes loom and then spins up the blockchain: "loom:init": "loom/loom init -f && loom/loom run".

Anyway, it's `loom:init`. I'll fix the typo.